pkg/prom/cluster/node.go

```diff
@@ -214,8 +206,8 @@ func (n *node) performClusterReshard(ctx context.Context, joining bool) error {
 			return nil, nil
 		}
 
-		ctx = user.InjectOrgID(ctx, "fake")
-		return nil, n.notifyReshard(ctx, id)
+		c = user.InjectOrgID(c, "fake")
```

Comment:
Using the parent ctx seemed like a legit issue, since it is not the same as the context passed in the function.